CONTRIBUTING.md

```diff
@@ -0,0 +1,166 @@
+- Check out open issues with the label `good first issue`.
+
+### Documentation
+
+- Fix typos, improve examples, or enhance explanations.
+
+## How to make a contribution?
+
+### Set up a fork of a repository
+
+1. On GitHub, navigate to the repository.
+1. In the top-right corner of the page, click Fork.
+1. Select an owner for the forked repository from the dropdown menu under "Owner".
+1. The fork will be named the same as the upstream repository as default. Optionally, to further distinguish your fork, type a name in the "Repository name" field.
+1. Optionally, type a description for your fork in the "Description" field.
+1. Optionally, select "Copy the `main` branch only".
+1. Click "Create fork".
+
+### Get your fork ready to start contributing and building
+
+The 'galasa' repository is divided into several modules that are stored within the `/modules` directory. Certain modules require artifacts from other modules as they have a dependency on them. The build workflow attempts to download these artifacts either from the same workflow run or from a previously successful completed workflow run. Therefore, before you can open a Pull Request on your forked repository from one branch to another, you will need to run an initial workflow run of the "Main Build Orchestrator".
+
+1. As default, GitHub Actions will be disabled on your fork. You will need to enable the GitHub Actions workflows by going to the Actions tab on your fork and clicking "I understand my workflows, go ahead and enable them".
+1. To run the "Main Build Orchestrator" on your fork, you will need to configure some repository secrets and variables. Follow the section below [Configure repository secrets and variables](#configure-repository-secrets-and-variables).
+1. Trigger an initial workflow run of the "Main Build Orchestrator". You can do this by going to the Actions tab, selecting "Main Build Orchestrator" from the list of workflows on the left (you may need to click "Show more workflows..."), then clicking "Run workflow" on the right.
+
+### Configure repository secrets and variables
+
+To run certain workflows on your fork, you will need to configure some repository secrets and variables so that the workflows can access them. These secrets and variables are used for things such as authenticating to the GitHub Container Registry and supplying Maven and Gradle builds with your GPG key information to sign artifacts.
+
+#### Repository secrets and variables that required configuring
+
+##### Repository variables
+
+To set a repository variable:
+1. Navigate to your repository settings.
+1. Select 'Secrets and variables', then 'Actions' from the menu.
+1. Click the 'Variables' tab.
+1. Select 'New repository variable'.
+1. Enter the variable name and value, then click 'Add variable'.
+
+**Required repository variables:**
+
+1. WRITE_GITHUB_PACKAGES_USERNAME: This requires the GitHub username you want to use to log into GitHub Container Registry.
```

Comment:
Is this checked against anything ? does it have to match anything ?

Reply:
Its used to authenticate to GHCR and will then be used as the credentials to push to your user's GitHub Packages so I guess yes it needs to match your user as so its allowed to push to your GHCR

Reply:
I'll rewrite this to something like `This requires your GitHub username so you can authenticate to GitHub Container Registry and push images to your personal packages.`